source/extensions/common/wasm/wasm_vm.h

```diff
@@ -18,36 +17,13 @@ namespace Extensions {
 namespace Common {
 namespace Wasm {
 
-/**
- * Wasm host stats.
- */
-#define ALL_VM_STATS(COUNTER, GAUGE)                                                               \
-  COUNTER(created)                                                                                 \
-  COUNTER(cloned)                                                                                  \
-  GAUGE(active, NeverImport)
-
-struct VmStats {
-  ALL_VM_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT)
-};
-
 // Wasm VM data providing stats.
 class EnvoyWasmVmIntegration : public proxy_wasm::WasmVmIntegration,
                                Logger::Loggable<Logger::Id::wasm> {
 public:
   EnvoyWasmVmIntegration(const Stats::ScopeSharedPtr& scope, absl::string_view runtime,
                          absl::string_view short_runtime)
-      : scope_(scope), runtime_(std::string(runtime)), short_runtime_(std::string(short_runtime)),
-        runtime_prefix_(absl::StrCat("wasm_vm.", short_runtime, ".")),
-        stats_(VmStats{ALL_VM_STATS(POOL_COUNTER_PREFIX(*scope_, runtime_prefix_),
-                                    POOL_GAUGE_PREFIX(*scope_, runtime_prefix_))}) {
-    stats_.created_.inc();
-    stats_.active_.inc();
-    ENVOY_LOG(debug, "WasmVm created {} now active", runtime_, stats_.active_.value());
-  }
-  ~EnvoyWasmVmIntegration() override {
-    stats_.active_.dec();
-    ENVOY_LOG(debug, "~WasmVm {} {} remaining active", runtime_, stats_.active_.value());
-  }
+      : scope_(scope), runtime_(std::string(runtime)), short_runtime_(std::string(short_runtime)) {}
```

Comment:
seems like we don't need to pass the scope here anymore?

Reply:
same goes for other members?

Reply:
Thanks, deleted them except runtime which has a getter method and is used.